packages/contracts/contracts/CdpManager.sol

```diff
@@ -442,6 +443,16 @@ contract CdpManager is CdpManagerStorage, ICdpManager, Proxy {
             sortedCdps.batchRemove(_toRemoveIds);
         }
 
+        // reinsert partially redemeed CDP if any
+        if (_cId != bytes32(0) && _partialRedeemedNewNICR > 0) {
```

Comment:
The Current (see above) will be reinserter as long as the partial logic was met:
- Partial, hint still valid
- Id is non-zero